api/controllers/UserController.js

```diff
@@ -346,44 +321,65 @@ module.exports = {
           ];
         }
         Promise.all(promises).then(function(result) {
-          res.json('ok', 200);
+          res.ok();
         }, function(error) {
           console.log(error);
-          res.json(error, 500);
+          res.status(500).json(error);
         });
       });
     });
   },
 
   setLocalBan: function (req, res) {
-    if (!req.user.isMod) {
-      res.json({error: "Not a mod"}, 403);
-      return;
-    }
-
     User.findOne(req.allParams().userId).exec(function (err, user) {
       if (!user) {
-        return res.json({error: "Can't find user"}, 404);
+        return res.notFound("Can't find user");
       }
 
       user.banned = req.allParams().ban;
       user.save(function (err) {
         if (err) {
-          return res.json(err, 500);
+          return res.serverError(err);
         }
-        res.json(user, 200);
+        res.ok(user);
       });
     });
   },
   
   bannedUsers: function (req, res) {
-    if (!req.user.isMod) {
-      res.json({error: "Not a mod"}, 403);
-      return;
-    }
-
     User.find({banned: true}).exec(function (err, users) {
-      res.json(users, 200);
+      res.ok(users);
     });
+  },
+
+  clearSession: function (req, res) {
+    Reddit.checkModeratorStatus(
+      sails.config.reddit.adminRefreshToken,
+      req.user.name,
+      'pokemontrades',
+      function(err, response) {
+        if (err) {
+          console.log('Failed to check whether /u/' + user.name + ' is a moderator.');
+          console.log(err);
+          return res.serverError();
+        }
+        if (response.data.children.length) { //User is a mod, clear session
+          User.findOne({name: req.allParams().name}, function (err, user) {
+            if (err) {
+              return res.serverError(err);
+            }
+            if (!user) {
+              return status(404).json("That user could not be found.");
+            }
+            Sessions.destroy({session: new RegExp('"user":"' + user.id + '"')}).exec(function (err) {
+              if (err) {
+                return res.serverError(err);
+              }
+              res.status(200).json("Successfully cleared /u/" + req.allParams().name + "'s sessions.");
```

Comment:
It doesn't seem to work when you clear your own session -- a new session is silently created in order to return a response properly. However, it clears other users' sessions properly.